app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt

```diff
@@ -96,19 +108,101 @@ open class FinanceViewModel(val tripsRepository: TripsRepository, val tripId: St
   /**
    * Updates the currency code of the current trip.
    *
-   * This method retrieves the current trip from the repository, updates it with the new currency
-   * code, and refreshes the related state lists.
+   * This method retrieves the current trip from the repository, updates the exchange rate, converts
+   * all expenses to the new currency, updates the trip with the new currency code, and refreshes
+   * the related state lists.
    *
-   * @param currencyCode The new currency code to be used for the trip.
+   * @param newCurrencyCode The new currency code to be used for the trip.
    */
-  open fun updateCurrency(currencyCode: String) {
+  open fun updateCurrency(newCurrencyCode: String) {
     viewModelScope.launch {
       val currentTrip = tripsRepository.getTrip(tripId)!!
-      tripsRepository.updateTrip(currentTrip.copy(currencyCode = currencyCode))
-      updateStateLists()
+      updateExchangeRate(currentTrip.currencyCode, newCurrencyCode)
+      if (exchangeRate.value != null) {
```

Comment:
You can put this if block inside of updateExchangeRate to not check again if it was successfull, since it's called "update". Or call it set or get

Reply:
Yes it makes more sens I will change the name of the method.